### PR TITLE
Add whitelist indicator for /seen and /whois

### LINKED ISSUE #5481: Whitelist indicator in /seen and in /whois (optional)
### Feature description

Hi, I was thinking about an easier way to check whether a player is whitelisted, as currently there is no built-in checker in vanilla, Paper or EssX. The easiest thing would be to add another line to the admin output of /seen. Along with the UUID, IP, location etc, there could also be an additional line - Whitelisted: true/false.

### How the feature is useful

When you want to check if a player is whitelisted, you can either run /whitelist list, or you can try to add/remove them and based on the result, you know. I think adding this piece of information into /seen (and/or /whois) could be a quicker and more practical way of checking.

Thank you!

### PR DESCRIPTION
**Pull request by tazaxer**
### Information

This PR closes #5481. 

### Details

**Proposed feature:**  

I've added a whitelist indicator on the commands whois and seen

When executing /whois the whitelist status of the target player will be displayed at the end as TRUE or FALSE

When executing /seen and if the target player is offline the whitelist status of the target player will be displayed at the end as TRUE or FALSE

**Environments tested:**    

OS: MacOS 13.4.1

Java version:  17.0.5

- [x] Most recent Paper version (1.XX.Y, git-Paper-BUILD)
- [x] CraftBukkit/Spigot/Paper 1.12.2
- [x] CraftBukkit 1.8.8

**Demonstration:**  

Before: https://imgur.com/a/mSFLzye
After: https://imgur.com/a/jiMvcmN

### CODE REVIEW

File: Essentials/src/main/java/com/earth2me/essentials/User.java [hunk 1]
```diff
@@ -1242,4 +1242,9 @@ public boolean isToggleShout() {
         }
         return toggleShout == null ? toggleShout = ess.getSettings().isShoutDefault() : toggleShout;
     }
+
+    public boolean isWhitelisted() {
```
Comment (JRoy) [inline Essentials/src/main/java/com/earth2me/essentials/User.java]: We don't need a new method here, just use user.getBase().isWhitelisted() in /seen and /whois

File: Essentials/src/main/resources/messages_en.properties [hunk 2]
```diff
@@ -1557,6 +1557,7 @@ whoisPlaytime=§6 - Playtime\:§r {0}
 whoisTempBanned=§6 - Ban expires\:§r {0}
 whoisTop=§6 \=\=\=\=\=\= WhoIs\:§c {0} §6\=\=\=\=\=\=
 whoisUuid=§6 - UUID\:§r {0}
+whoisWhitelist=§6 - Whitelist\:§r {0}
```
Comment (JRoy) [inline Essentials/src/main/resources/messages_en.properties]: This needs to go in messages.properties, not messages_en.properties

File: Essentials/src/main/java/com/earth2me/essentials/commands/Commandwhois.java [hunk 3]
```diff
@@ -71,6 +71,8 @@ public void run(final Server server, final CommandSource sender, final String co
             sender.sendMessage(tl("whoisMutedReason", user.isMuted() ? muteTimeout > 0 ? DateUtil.formatDateDiff(muteTimeout) : tl("true") : tl("false"),
                 user.getMuteReason()));
         }
+
+        sender.sendMessage(tl("whoisWhitelist", user.isWhitelisted() ? tl("true") : tl("false")));
```
Comment (JRoy) [inline Essentials/src/main/java/com/earth2me/essentials/commands/Commandwhois.java]: i'd prefer this is moved below the `whoisSpeed` line

File: Essentials/src/main/java/com/earth2me/essentials/commands/Commandseen.java [hunk 4]
```diff
@@ -188,6 +189,10 @@ private void seenOffline(final CommandSource sender, final User user, final bool
                 sender.sendMessage(tl("whoisLocation", loc.getWorld().getName(), loc.getBlockX(), loc.getBlockY(), loc.getBlockZ()));
             }
         }
+
+        if (showWhitelist) {
+            sender.sendMessage(tl("whoisWhitelist", user.isWhitelisted() ? tl("true") : tl("false")));
+        }
```
Comment (JRoy) [inline Essentials/src/main/java/com/earth2me/essentials/commands/Commandseen.java]: move this above line 154

**Review by JRoy — CHANGES_REQUESTED** [review]
few things, overall good

File: Essentials/src/main/resources/messages.properties [hunk 5]
```diff
@@ -1569,6 +1569,7 @@ whoisPlaytime=\u00a76 - Playtime\:\u00a7r {0}
 whoisTempBanned=\u00a76 - Ban expires\:\u00a7r {0}
 whoisTop=\u00a76 \=\=\=\=\=\= WhoIs\:\u00a7c {0} \u00a76\=\=\=\=\=\=
 whoisUuid=\u00a76 - UUID\:\u00a7r {0}
+whoisWhitelist=§6 - Whitelist\:§r {0}
```
Comment (JRoy) [inline Essentials/src/main/resources/messages.properties]:
```suggestion
whoisWhitelist=\u00a76 - Whitelist\:\u00a7r {0}
```

**Review by JRoy — APPROVED** [review]

**Comment by JRoy** [comment]
Thanks

**Comment by takatalvi** [comment]
Thank you so much for this!

**Comment by Bobcat00** [comment]
You guys are amazing:
https://github.com/EssentialsX/Essentials/pull/4618#issuecomment-974799094

**Pull request merged**